core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataDeleteIndexService.java

```diff
@@ -66,9 +66,11 @@ public MetaDataDeleteIndexService(Settings settings, ThreadPool threadPool, Clus
         this.nodeIndexDeletedAction = nodeIndexDeletedAction;
     }
 
-    public void deleteIndex(final Request request, final Listener userListener) {
-        final DeleteIndexListener listener = new DeleteIndexListener(userListener);
-        clusterService.submitStateUpdateTask("delete-index [" + request.index + "]", Priority.URGENT, new ClusterStateUpdateTask() {
+    public void deleteIndices(final Request request, final Listener<Response> userListener) {
+        Collection<String> indices = Arrays.asList(request.indices);
+        final DeleteIndexListener deleteListener = new DeleteIndexListener(userListener);
+
+        clusterService.submitStateUpdateTask("delete-index [" + indices + "]", Priority.URGENT, new ClusterStateUpdateTask() {
```

Comment:
I think we might end up with double square brackets here given that the string representation of the list will be like 
"[index1, index2, index3]" already.